lib/msal-node/src/client/ManagedIdentitySources/BaseManagedIdentitySource.ts

```diff
@@ -166,4 +182,54 @@ export abstract class BaseManagedIdentitySource {
             managedIdentityRequest
         );
     }
+
+    public getManagedIdentityUserAssignedIdQueryParameterKey(
+        managedIdentityIdType: ManagedIdentityIdType
+    ): string {
+        switch (managedIdentityIdType) {
+            case ManagedIdentityIdType.USER_ASSIGNED_CLIENT_ID:
+                this.logger.info(
+                    "[Managed Identity] Adding user assigned client id to the request."
+                );
+                return ManagedIdentityUserAssignedIdQueryParameterNames.MANAGED_IDENTITY_CLIENT_ID;
+
+            case ManagedIdentityIdType.USER_ASSIGNED_RESOURCE_ID:
+                this.logger.info(
+                    "[Managed Identity] Adding user assigned resource id to the request."
+                );
+                return ManagedIdentityUserAssignedIdQueryParameterNames.MANAGED_IDENTITY_RESOURCE_ID;
+
+            case ManagedIdentityIdType.USER_ASSIGNED_OBJECT_ID:
+                this.logger.info(
+                    "[Managed Identity] Adding user assigned object id to the request."
+                );
+                return ManagedIdentityUserAssignedIdQueryParameterNames.MANAGED_IDENTITY_OBJECT_ID;
+            default:
+                throw createManagedIdentityError(
+                    ManagedIdentityErrorCodes.invalidManagedIdentityIdType
+                );
+        }
+    }
 }
+
+export const getValidatedEnvVariableUrlString = (
+    envVariableStringName: string,
+    envVariable: string,
+    sourceName: string,
+    logger: Logger
+): string => {
+    try {
+        return new UrlString(envVariable).urlString;
```

Comment:
What is `UrlString` ? How is it different than `Url`?

Reply:
It's MSALJS's own URL utils class.

"Url object class which can perform various transformations on url strings."

We use it to validate urls